aip/general/0130.md

```diff
@@ -0,0 +1,88 @@
+| Stateless Methods                  | A method that has no permanent effect on any data within the API (e.g. translating text)                                                                                                                      | [AIP-136][]                                        | unused or handwritten           | automatable     | automtable     | automatable     |
+| **None of the above**              |                                                                                                                                                                                                               |                                                    |                                 |                 |                |                 |
+| Streaming methods                  | Methods that communicate via client, server, or bi-directional streams.                                                                                                                                       |                                                    | handwritten                     | handwritten     | handwritten    | automatable     |
+
+### Choosing a method category
+
+While designing a method, API authors should choose from the defined categories
+in the following order:
+
+1. Standard methods (on collections and resources)
+1. Standard batch or aggregate methods
+1. Custom methods (on collections, resources, or stateless)
+1. Streaming methods
+
+## Rationale
+
+Resource-oriented standard and custom methods are recommended first, as they can
+be expressed in the widest variety of clients (IaC, CLIs, UIs, and so on), and
+offer the most uniform experience that allows users to apply their knowledge of
+one API to another.
+
+If a standard method is unsuitable, then custom methods (that are mounted to a
+resource or collection) offer a lesser, but still valuable level of consistency,
+helping the user reason about the scope of the action and the object whose
+configuration is read to inform that action. Although mutative custom methods
+are not uniform enough to have a automated integration with exclusively
+resource-oriented clients such as [IaC][] clients, they are still a pattern that
+can easily recognized by CLIs, UIs, and SDKs.
+
+If one cannot express their APIs in a resource-oriented fashion at all, then the
+operation falls in a category where the lack of uniformity makes it difficult
+for any client aside from SDKs to model the operation. This category is
+preferred last due to the fact that a user cannot rely on their knowledge of
+similar APIs, as well as the issue that integration with many clients will
+likely have to be hand-written.
+
+[AIP-121]: ./0121.md
+[AIP-132]: ./0132.md
+[AIP-133]: ./0133.md
+[AIP-134]: ./0134.md
+[AIP-135]: ./0135.md
+[AIP-136]: ./0136.md
+[AIP-159]: ./0159.md
+[AIP-231]: ./0231.md
```

Comment:
AIP 162 is missing. It's easy to see if you view the markdown.

Reply:
ack, thanks!